server/src/main/kotlin/com/xebia/functional/xef/server/exceptions/ExceptionsHandler.kt

```diff
@@ -0,0 +1,32 @@
+package com.xebia.functional.xef.server.exceptions
+
+import com.xebia.functional.xef.server.models.exceptions.XefExceptions
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.plugins.statuspages.*
+import io.ktor.server.response.*
+
+fun Application.exceptionsHandler() {
+    install(StatusPages) {
+        exception<Throwable> { call, cause ->
+            if (cause is XefExceptions) {
+                call.manageException(cause)
+            } else {
+                call.respond(HttpStatusCode.InternalServerError, cause.message ?: "Unexpected error")
+            }
+            call.respondText(cause.localizedMessage, status = HttpStatusCode.InternalServerError)
```

Comment:
Why do we respond twice here? Once inside the if-else branch and once after that.

Reply:
If I'm not wrong, they belong to two different branches. 
```kotlin
install(StatusPages) {
        exception<Throwable> { call, cause ->
            if (cause is XefExceptions) {
                call.manageException(cause)
            } else {
                call.respond(HttpStatusCode.InternalServerError, cause.message ?: "Unexpected error")
            }
            call.respondText(cause.localizedMessage, status = HttpStatusCode.InternalServerError)
        }
        status(HttpStatusCode.NotFound) { call, status ->
            call.respondText(text = "404: Page Not Found", status = status)
        }
    }
```

The first is the `exception` handler. In which we are getting the exceptions and in the other case is the `status`, in which is changing the response of all the NotFound responses.

Reply:
I mistake 😓 Thanks!